jazz_codeq/test/test.js

```diff
@@ -0,0 +1,757 @@
+					expect(result.branch).to.eq("master");
+					sinon.assert.calledOnce(requestPromiseStub);
+					requestPromiseStub.restore();
+				});
+			});
+
+			it('should return error when response statusCode is not 200', () => {
+				const response = { statusCode: 400, body: { message: "error"} };
+				const body = { data: {} };
+				const requestPromiseStub = sinon.stub(request, "Request").callsFake(obj => {
+					return obj.callback(null, response, body);
+				});
+
+				const expectedResult = utils.getProjectBranch("token", query, config)
+				.catch(err => {
+					expect(err.code).to.eq(400);
+					sinon.assert.calledOnce(requestPromiseStub);
+					requestPromiseStub.restore();
+				});
+			});
+		});
+
+		describe('getCodeqReport tests', () => {
+			let config = {};
+			let sandbox;
+			let query = {
+				domain: "domain",
+				service: "service"
+			};
+			let metrics = ['security'];
+			
+			beforeEach(() => {
+				sandbox = sinon.createSandbox();
+				config = {
+					"SONAR_PROJECT_KEY": "jazz",
+					"SONAR_URL": "sonarurl",
+					"METRIC_MAP": {
+						"security": "vulnerabilities",
+						"code-coverage": "coverage",
+						"code-smells": "code_smells",
+						"lines-of-code": "ncloc",
+						"files": "files",
+						"vulnerabilities": "high_severity_vulns"
+					},
+					"SERVICE_API_URL": "serviceurl",
+					"HELP_SERVICE": "/helpurl"
+				};
+			});
+	
+			afterEach(() => {
+				sandbox.restore();
+				config = {};
+				query: {};
+			});
+	
+			it('should return report name and link when statusCOde is 200', () => {
+				const response = { statusCode: 200 };
+				const body = { 
+					data: {},
+					sonarMeasures: [{ 
+						metric: 'vulnerabilities',
+						history: [{'date': 'date1'}]
+					}, 
+					{
+						metric: 'coverage',
+						history: [{'date': 'date2'}]
+					}]
+				};
+				const requestPromiseStub = sinon.stub(request, "Request").callsFake(obj => {
+					return obj.callback(null, response, body);
+				});
+
+				const expectedResult = utils.getCodeqReport(metrics, "master", "authToken", "todate", "fromdate", query, config)
+				.then(result => {
+					expect(result.metrics.length).to.eq(1);
+					expect(result.metrics[0].link).to.eq('serviceurl/helpurl?metrics=security');
+					sinon.assert.calledOnce(requestPromiseStub);
+					requestPromiseStub.restore();
+				});
+			});
+
+			it('should return error when response statusCode is not 200', () => {
+				const response = { statusCode: 400, body: { errors: [{ msg: "error" }]} };
+				const body = { data: {} };
+				const requestPromiseStub = sinon.stub(request, "Request").callsFake(obj => {
+					return obj.callback(null, response, body);
+				});
+
+				const expectedResult = utils.getCodeqReport(metrics, "master", "authToken", "todate", "fromdate", query, config)
+				.catch(err => {
+					expect(err.code).to.eq(400);
+					expect(err.report_error).to.eq("error");
+					sinon.assert.calledOnce(requestPromiseStub);
+					requestPromiseStub.restore();
+				});
+			});

```

Comment:
How about test case when SonarQube doesn't return data?

Reply:
That is what the test is doing. Its looking at the error scenario. Not sure what is missing here?